adoc/chapters/programming_interface.adoc

```diff
@@ -3824,46 +3824,40 @@ a@
 ----
 void wait()
 ----
-   a@ Wait for the event and the command associated with
-      it to complete.
+   a@ Blocks until all commands associated with this event and any dependent
+      events have completed.
 
 a@
 [source]
 ----
 void wait_and_throw()
 ----
-   a@ Wait for the event and the command associated with
-      it to complete.
+   a@ Blocks until all commands associated with this event and any dependent
+      events have completed.
 
-Any unconsumed <<async-error,asynchronous errors>> from any context that the event
-was waiting on executions from will be passed to the
-<<async-handler>> associated with the context.
-If no user defined [code]#async_handler# is associated with
-the context, then an implementation-defined
-default <<async-handler>> is called to handle any errors, as
-described in <<subsubsec:exception.nohandler>>.
+Any unconsumed <<async-error,asynchronous errors>> which were produced by the
+execution of commands associated with this event or any dependent events, are
+passed to the <<async-handler>> associated with the respective context.
+
+If no user defined <<async-handler>> is associated with the context, then an
+implementation-defined default <<async-handler>> is invoked to handle errors,
+as described in <<subsubsec:exception.nohandler>>.
```

Comment:
I think this change is an improvement, but there are a couple other things that we might want to improve also.

The wording about "associated with the respective context" is not quite accurate because the async errors could also be reported to the handler that is associated with the queue.  In fact, [section 4.13.1.3](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_priorities_of_async_handlers) states that the handler associated with the queue has precedence over the handler associated with the context.  Perhaps we could say something like this instead:

> ... are passed to the application's async-handler as described in section 4.13.1.3.

I also wonder if the part about "which were produced by the execution of commands associated with this event or any dependent events" is accurate.  [Section 4.13.1.1](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsubsec:exception.async) says only that "any unconsumed asynchronous errors associated with the queue or context" are reported.  It's not clear to me what precisely we intended for a situation like this:

```
sycl::queue q{async_handler};
auto e1 = q.submit(/*....*/);
auto e2 = q.submit(/*...*/);  // Does *not* depend on e1
q.wait();  // Does not report async errors

e2.wait_and_throw();
```

Let's suppose the command associated with `e1` generated an async error.  Will this error be reported when `e2.wait_and_throw` is called?  I think there are three reasonable answers:

1. The spec requires that the error is *not* reported because the error is not generated by the command associated with `e2` or by any of its dependents.
2. The spec allows the error to be reported, but does not require it.
3. The spec requires that the error *is* reported because the call to `q.wait` ensures that the async error was generated, so it becomes an "unconsumed error" on the queue, and the call to `e2.wait_and_throw` is guaranteed to report any unconsumed errors.

Reply:
In the WG meeting today, the WG seems to be in favor of option (2).  We think the spec wording should be clarified to indicate this.

Reply:
These clarifications have been addressed in https://github.com/KhronosGroup/SYCL-Docs/pull/520/commits/3df6485b1248220252663a2a775517b247169147.